### ISSUE TITLE
Different Magic attribute value than to Network leads to errors when starting consensus

### ISSUE THREAD

**Issue by gsmachado** [neo-project/neo-modules#547]
**Describe the bug**

When there's a mismatch of `Magic` and `Network` attributes, the consensus can't start.

Probably this is intended. In this case, however, better error messages should be provided.

Or, if this is not intended, can `Magic` be set to a value and `Network` to another one (in all Plugins of `neo-modules`)?

Another thing, is: `Magic` attribute in the `ProtocolConfiguration` is still necessary with the introduction of `Network`? If yes, for which use cases?

**To Reproduce**

If `Magic` is set to `769` in the `ProtocolConfiguration` but all other `Network` attributes are set to `5195086` (in all plugins):

![image](https://user-images.githubusercontent.com/2124276/110707145-2b67e100-81f9-11eb-9230-7c76eb34835b.png)

If `Magic` is set to `5195086` in the `ProtocolConfiguration` but the wallet is not re-generated (i.e., `wallet.json` was generated still with `769`):

![image](https://user-images.githubusercontent.com/2124276/110707023-fc516f80-81f8-11eb-9694-af5988d1e250.png)

**Platform:**
 - `neo-node` on commit `a636e8b`
 - `neo-modules` on commit `b54fb54`
 - but, both pointing to `3.0.0-CI01238`

**Comment by gsmachado** [comment]
@superboyiii @cloud8little can you or someone else check this one?

**Comment by cloud8little** [comment]
> @superboyiii @cloud8little can you or someone else check this one?

> When there's a mismatch of Magic and Network attributes, the consensus can't start.
Yes, it is intended, when the network id mismatch, the plugin won't work. for all plugins, the same.

Different network configuration is used to connect to different Neo instance. It will used for NEOFs network in later design.

**Comment by gsmachado** [comment]
So, the `Magic` and `Network` attributes **SHOULD** be the same?

If yes:

- Why do they have different names?
- Why `Magic` attribute is not extinguished and only `Network` is used?

If no:

- Why a different `Magic` and `Network` attributes affect neo-node functionality?

**Comment by ixje** [comment]
I recently ran into this again, I think there should be a more friendly error message when the magic/network versions mismatch than "Object reference not set to an instance of of an object".
